sunpy/database/tests/test_database.py

```diff
@@ -440,6 +440,17 @@ def test_add_entry_from_hek_qr(database):
     # *something*
     assert len(database) > 1
 
+@pytest.mark.remote_data
+def test_download_from_hek_query_result(database, hek_qr, tmpdir):
+    assert len(database) == 0
+    database.download_from_hek_query_result(
+        hek_qr, path=str(tmpdir.join('{file}.fits')))
+    fits_pattern = str(tmpdir.join('*.fits'))
+    num_of_fits_headers = sum(
+        len(fits.get_header(file)) for file in glob.glob(fits_pattern))
+    assert len(database) == num_of_fits_headers > 0
+    for entry in database:
+        assert os.path.dirname(entry.path) == str(tmpdir)
```

Comment:
I'm not fun of `assert`s on loops. That stops the test on the first failure in the loop, while you may have more. If `tmpdir` should be the same in all, I would do:
```python
assert all([os.path.dirname(entry.path) == str(tmpdir) for entry in database)
```
or if we want that the output is more meaningful as in which directory is wrong:
```python
assert [os.path.dirname(entry.path) for entry in database] == [str(tmpdir)] * len(database)
```